src/components/AnonymousChat/child/ChatItemTargetText.tsx

```diff
@@ -1,100 +1,157 @@
-    flex: 1
-  },
-  userText: {
-    fontFamily: fonts.inter[600],
-    fontSize: 12,
-    lineHeight: 19.36
-  },
-  text: {
-    fontFamily: fonts.inter[400],
-    fontSize: 16,
-    lineHeight: 19.36
-  },
-  avatar: {
-    width: 24,
-    height: 24,
-    borderRadius: 12,
-    marginRight: 8
-  },
-  dot: {
-    width: 3,
-    height: 3,
-    borderRadius: 2,
-    marginLeft: 5,
-    marginRight: 5,
-    backgroundColor: colors.black,
-    alignSelf: 'center'
-  },
-  timeText: {
-    fontFamily: fonts.inter[200],
-    fontSize: 10,
-    lineHeight: 12.19,
-    alignSelf: 'center'
-  },
-  mr8: {
-    marginRight: 8
-  }
-});
+export const replyIcon = () => {
+  return (
+    <View style={styles.containerReply}>
+      <View style={styles.containerReplyIcon}>
+        <Icon name={'reply'} size={20} color={colors.black} />
+      </View>
+    </View>
+  );
+};
 
 const ChatItemTargetText = ({
   username = 'Anonymous User',
   time = '4h',
   isContinuous = false,
   message = '',
-  avatar
+  avatar,
+  chatType,
+  messageType,
+  data
 }: ChatItemMyTextProps) => {
+  const {
+    setReplyPreview,
+    onContextMenuPressed,
+    pulseAnimation,
+    animatedBubbleStyle,
+    animatedPulseStyle
+  } = useMessageHook();
+  const scrollContext = React.useContext(ScrollContext);
+  const swipeableRef = React.useRef<Swipeable>(null);
+
+  React.useEffect(() => {
+    if (scrollContext?.selectedMessageId === data?.id) {
+      pulseAnimation.value = withSequence(
+        withDelay(300, withTiming(1.1, {duration: 200})),
+        withTiming(1, {duration: 200})
+      );
+      scrollContext?.setSelectedMessageId(null);
+    }
+  }, [scrollContext?.selectedMessageId]);
+
+  const isReply = messageType === MESSAGE_TYPE_REPLY;
+  const isReplyPrompt = messageType === MESSAGE_TYPE_REPLY_PROMPT;
+  const isShowUserInfo = !isContinuous || isReplyPrompt || isReply;
+
+  const contextMenuActions: ContextMenuAction[] = [
+    {title: CONTEXT_MENU_REPLY, systemIcon: 'arrow.turn.up.left'}
+  ];
+
+  if (messageType !== MESSAGE_TYPE_DELETED) {
+    contextMenuActions.push({title: CONTEXT_MENU_COPY, systemIcon: 'square.on.square'});
+  }
+
   const renderAvatar = React.useCallback(() => {
-    if (isContinuous) return <View style={styles.avatar} />;
-    return <View style={styles.mr8}>{avatar}</View>;
-  }, []);
+    if (!isShowUserInfo) return <View style={styles.avatar} />;
+    return <View style={styles.mrBuble}>{avatar}</View>;
+  }, [isShowUserInfo, avatar]);
+
+  const onSwipeToReply = React.useCallback(
+    (direction: 'left' | 'right') => {
+      if (direction === 'right') return;
+      if (swipeableRef.current) swipeableRef.current?.close();
+      setReplyPreview({
+        id: data?.id ?? data?.message?.id,
+        user: {username},
+        message,
+        message_type: messageType as MessageType,
+        updated_at: time,
+        chatType
+      });
+    },
+    [data, username, time, message, messageType, chatType]
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

+ try {
    if (direction === 'right') return;
    if (swipeableRef.current) swipeableRef.current?.close();
    setReplyPreview({
      id: data?.id ?? data?.message?.id,
      user: {username},
      message,
      message_type: messageType as MessageType,
      updated_at: time,
      chatType
    });
+ } catch (error) {
+   console.error('Failed to handle swipe to reply:', error);
+ }



<!-- This is an auto-generated comment by OSS CodeRabbit -->